API-Samples/Java/app/src/main/java/videoindexersamples/VideoIndexerClient.java

```diff
@@ -0,0 +1,239 @@
+        System.out.printf("Searching videos in account %s for video ID %s.\n", account.properties.accountId, videoId);
+
+        Map<String, String> map = new HashMap<>();
+        map.put("accessToken", this.accountAccessToken);
+        map.put("id", videoId);
+        var queryParam = Utils.toQueryParamString(map);
+
+        try {
+            var requestUri = String.format("%s/%s/Accounts/%s/Videos/Search?%s", ApiUrl, account.location, account.properties.accountId, queryParam);
+            return httpStringResponse(Utils.httpGetRequest(requestUri)).body();
+        } catch (Exception ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    /**
+     * Gets an account. Calls the getAccount API (<a href="https://github.com/Azure/azure-rest-api-specs/blob/main/specification/vi/resource-manager/Microsoft.VideoIndexer/stable/2022-08-01/vi.json#:~:text=%22/subscriptions/%7BsubscriptionId%7D/resourceGroups/%7BresourceGroupName%7D/providers/Microsoft.VideoIndexer/accounts/%7BaccountName%7D%22%3A%20%7B">...</a>)
+     *
+     * @return : The Account Info ( accountId and Location) if successful, otherwise throws an exception</returns>
+     */
+    public Account getAccountInfo() {
+        System.out.println("Getting Account Info ( Location/AccountId)");
+        try {
+            var requestUri = String.format("%s/subscriptions/%s/resourcegroups/%s/providers/Microsoft.VideoIndexer/accounts/%s?api-version=%s", AzureResourceManager, SubscriptionId, ResourceGroup, AccountName, ApiVersion);
+            try {
+                HttpRequest request = HttpRequest.newBuilder().uri(new URI(requestUri))
+                        .headers("Content-Type", "application/json;charset=UTF-8")
+                        .headers("Authorization", String.format("Bearer %s", this.armAccessToken))
+                        .GET()
+                        .build();
+                var responseBodyJson = httpStringResponse(request).body();
+                this.account = gson.fromJson(responseBodyJson, Account.class);
+            } catch (URISyntaxException | IOException | InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        } catch (Exception ex) {
+            throw new RuntimeException(ex);
+        }
+        return this.account;
+    }
+
+    /**
+     * Calls getVideoIndex API in 10 second intervals until the indexing state is 'processed'(<a href="https://api-portal.videoindexer.ai/api-details#api=Operations&operation=Get-Video-Index">...</a>)
+     *
+     * @param videoId - the Id of the video being uploaded
+     * @return : true if the wait operation succedded , false otherwise
+     */
+    public boolean waitForIndex(String videoId) {
+        System.out.printf("Waiting for Video %s to finish indexing.\n", videoId);
+
+        Map<String, String> map = new HashMap<>();
+        map.put("accessToken", this.accountAccessToken);
+        map.put("language", "English");

```

Comment:
I think you might want to leave this empty, this way, you will get the default (indexed langauge)

Reply:
the author of the original sample want to reflect the possibility you can use on the API